pkg/controller/node/node_controller.go

```diff
@@ -763,28 +785,74 @@ func (ctrl *Controller) handleErr(err error, key interface{}) {
 // 2. If a MachineConfig changes, we should wait for the OS image build to be
 // ready so we can update both the nodes' desired MachineConfig and desired
 // image annotations simultaneously.
+
+func (ctrl *Controller) GetConfigAndBuild(pool *mcfgv1.MachineConfigPool) (*mcfgv1alpha1.MachineOSConfig, *mcfgv1alpha1.MachineOSBuild, error) {
```

Comment:
quibble: If this function is not intended to be used outside of the NodeController, it should be renamed to `getConfigAndBuild()` because of Go's export rules.